src/pydicom/dataset.py

```diff
@@ -2527,7 +2689,7 @@ def __setattr__(self, name: str, value: Any) -> None:
             # XXX note if user misspells a dicom data_element - no error!!!
             object.__setattr__(self, name, value)
 
-    def _set_file_meta(self, value: Optional["Dataset"]) -> None:
+    def _set_file_meta(self, value: "Dataset | None") -> None:
```

Comment:
Hm, would have expected `"Dataset" | None` for the type hint, but I may be wrong.

Reply:
Weird, I could've sworn you couldn't mix them like that, but I tested it and it works OK. It's possible I was using Python < 3.10, though.

Reply:
Ah ok, didn't know that, thanks!